community/kernel/src/main/java/org/neo4j/kernel/impl/store/TemporalType.java

```diff
@@ -244,22 +245,18 @@ public Value decodeForTemporal( long[] valueBlocks, int offset )
                     }
                     else
                     {
-                        throw new UnsupportedOperationException( "Cannot yet store DateTime with ZoneID" );
+                        int nanoOfSecond = (int) (valueBlocks[offset] >>> 33);
+                        long epochSecond = valueBlocks[1 + offset];
+                        short zoneNumber = (short) valueBlocks[2 + offset];
+                        return DateTimeValue.datetime( epochSecond, nanoOfSecond,
+                                ZoneId.of( TimeZoneMapping.TIME_ZONE_SHORT_TO_STRING.get( zoneNumber ) ) );
                     }
                 }
 
                 @Override
                 public int calculateNumberOfBlocksUsedForTemporal( long firstBlock )
                 {
-                    if ( storingZoneOffset( firstBlock ) )
-                    {
-                        return 3;
-                    }
-                    else
-                    {
-                        // TODO proper number
-                        return 3;
-                    }
+                    return 3;
```

Comment:
A little outline of what's in these 3 blocks as a comment would be dandy

Reply:
I will put that into the comment in PropertyStore, where all store layouts are summarized. Thanks for the reminder!